### PR TITLE
[5/6][kimi-deterministic] Compute decode logprobs via log_softmax under deterministic inference

### PR DESCRIPTION
**Pull request by zyzshishui**
> **Stacked series (5/6)**: stacked on `kimi-deterministic/4-dp-attention-sampling-positions` (#30820), so the diff below includes 1-4/6. This PR's own changes are the last 2 commits (ending 7ab0ae38a). Combined CI: #30822.

## Motivation

At `temperature > 0` with `return_logprob`, decode logprobs come from `torch.log(torch.softmax(logits / T))` while prefill rescoring uses `F.log_softmax`; the kernels round differently (~1e-6), breaking bitwise prefill/decode alignment in deterministic mode. (Part 5/6 of the kimi-deterministic series.)

## Modifications

Under deterministic inference with `return_logprob`, compute returned logprobs via `F.log_softmax` on the temperature-scaled logits; sampling unchanged (the greedy path already did this).

## Accuracy Tests

Kimi-K2-Instruct-BF16, 16 GPUs (TP16 / DP8-attention / EP16, fa4 + deep_gemm + deepep normal, radix on), 128 prompts (1k-16k) x 512 tokens, abs tol 0.0: replay, shuffled concurrency-32, bs1 vs bs128, cached-prefix extend vs cold prefill, decode logprobs vs prefill rescoring (65,536 tokens) — all bitwise identical.

The prefill/decode alignment phase failed at temperature 1.0 before this change.

## Speed Tests and Profiling

One extra `log_softmax` only under deterministic inference with `return_logprob` at `temperature > 0`.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #29979964802](https://github.com/sgl-project/sglang/actions/runs/29979964802)
Latest PR Test (Extra): :x: [Run #29979964709](https://github.com/sgl-project/sglang/actions/runs/29979964709)

### CODE REVIEW

File: python/sglang/srt/layers/sampler.py [hunk 1]
```diff
@@ -176,6 +176,16 @@ def forward(
                 # Standard path: do softmax and sample from probs.
                 logits.div_(sampling_info.temperatures)
 
+                if (
```
Comment (ch-wan) [inline python/sglang/srt/layers/sampler.py:194]: **[nit]** The new branch has no comment explaining *why* deterministic mode recomputes logprobs via `log_softmax` after temperature scaling. Nearby RL code documents its matching goal; without a similar note, a later cleanup could re-collapse this to `torch.log(probs)` and silently re-break prefill/decode bitwise alignment (~1e-6).

**Suggestion:** Add a one-line comment, e.g. that under deterministic inference returned logprobs must use `F.log_softmax` (same kernel as prefill rescoring) rather than `log(softmax)`, because the two disagree at ~1e-6 even when mathematically equivalent.

**Review by ch-wan — APPROVED** [review]
